src/errors/test-run/templates.js

```diff
@@ -379,4 +379,8 @@ export default {
     [TEST_RUN_ERRORS.multipleWindowsModeIsNotSupportedInRemoteBrowserError]: err => `
         Multi window mode is supported in local browsers only. Run tests locally to use the "${err.methodName}" method.
     `,
+
+    [TEST_RUN_ERRORS.cannotCloseWindowWithoutParent]: () => `
+        Cannot close a window without parent.
```

Comment:
I feel that this message is quite technical. Maybe we can explain the situation better
```
Cannot close the window. Likely it is the original window or a window which parent was already closed. Try to remove this `closeWindow` call or consider submitting a bug report: https://github.com/DevExpress/testcafe/issues/new?template=bug-report.md
``` 
```

Reply:
`Cannot close the window because it does not have a parent. The parent window has already been closed or you are attempting to close the root browser window where tests were launched.`

Reply:
Do we really need that part about removing `closeWindow` (which seems obvious) and reporting a bug (do we expect it to indicate bugs?)

Reply:
'The parent window has already been closed'
We do not allow do close window if it has children

Reply:
I agree with @VasilyStrelyaev. Not sure, that we need part about reporting a bug

Reply:
Parent can be closed by on-page scripts.

Reply:
```js
fixture`dtf`.page`dtf.ru`;

test('check terms&conditions', async t => {
    await t
        .click('body > div.main_menu.layout > div.layout__content.l-flex > div.main_menu__right-side.l-ml-auto.lm-ml-0.l-flex.l-fa-center > div.main_menu__auth.l-ml-10.lm-ml-10.l-mr-40.lm-mr-0 > div > span')
        .click('body > div.main_menu.layout > div.popup > div.popup__container.popup__container--shown > div > div.popup__container__window__tpl > div > div > div.auth-form__main > div.auth-form__tab.auth_tab.auth_tab--signin-start > div.auth-form__content > div.social-auth.social-auth--with-email > div:nth-child(5)')
        // Currently can't click on the link, since it opens in a new tab but instead of a new window
        //.click('#bd > div > div.app-info > dl > dd:nth-child(10) > p > a')
        .openWindow('https://dtf.ru/terms')

    const window = await t.getCurrentWindow();

    // Terms and conditions title
    console.log(await t.eval(() => document.title));

    await t
        .switchToPreviousWindow()
        .click('#cancel')
        .click('#bd > p.cancel-callback > a');

    // The twitter login page is now closed. The terms and conditions page is still opened, and has no parent.
  
    //The landing title
    console.log(await t.eval(() => document.title));
    
    // We cannot switch to the page without a parent  
    await t.switchToWindow(window);
});
```

Reply:
You are right. Thanks. We do not allow to close the window if it has children only if we use API.

Reply:
Checked with Dirk:

Cannot close the window because it does not have a parent. The parent window was closed or you are attempting to close the root browser window where tests were launched.